packages/indexer/src/plugins/persistence.ts

```diff
@@ -0,0 +1,89 @@
+import type { Cursor } from "@apibara/protocol";
+import { type Database, type ISqlite, open } from "sqlite";
+import { defineIndexerPlugin } from "./config";
+
+type SqliteArgs = ISqlite.Config;
+
+export function sqlitePersistence<TFilter, TBlock, TRet>(args: SqliteArgs) {
+  return defineIndexerPlugin<TFilter, TBlock, TRet>((indexer) => {
+    let db: Database;
+
+    indexer.hooks.hook("run:before", async () => {
+      db = await open(args);
+      await SqlitePersistence.initialize(db);
+    });
+
+    indexer.hooks.hook("connect:before", async ({ request }) => {
+      const store = new SqlitePersistence(db);
```

Comment:
I think it's possible to instantiate the persistence once and share the instance between the hooks (like you do with `db`).